src/main/java/io/stargate/sgv2/jsonapi/config/OperationsConfig.java

```diff
@@ -100,6 +100,23 @@ public interface OperationsConfig {
   @WithDefault("1000")
   int maxVectorSearchLimit();
 
+  /**
+   * @return Maximum size of keys read from database to return count, Setting it to -1 will use
+   *     Cassandra's count function. Default is 1000 <code>1000</code> command.
+   */
+  @Max(10000)
+  @WithDefault("1000")
+  int maxCountLimit();
+
+  /**
+   * @return Defines the default page size for count operation, having separate config because count
+   *     will more keys per page, defaults to <code>100</code>.
```

Comment:
nit: "will more" -> "will read more". Also separate from which setting? (add link/ref to the other setting)